events/templates/event_detail.html

```diff
@@ -3,27 +3,49 @@
 <div class="card my-3">
   <div class="card-header">
     <h2 class="d-inline">
-      <a href="{%url 'event detail' event.id%}"> {{event.name|capfirst}} </a>
+      <a href="{% url 'event detail' event.id %}"> {{event.name|capfirst}} </a>
     </h2>
-    <h4 class="d-inline">by {{event.author}}</h4>
+    <h4 class="d-inline">by
+      <a href="{% url 'profile' event.author.id %}">{{event.author}}</a>
+      <img src="{{ event.author.profile.image.url }}" class="rounded-circle" height="40px" alt="">
+    </h4>
     <h3>{{event.category}}</h3>
 
     {% if event.author.id == user.id %}
       <a class="btn btn-primary" href="{%url 'event edit' event.id%}">Edit</a>
-    
+    {% endif %}
+
+    {% if user not in event.signed_up.all %}
+      <a class="btn btn-secondary" href="{% url 'event signup' event.id %}">Signup</a>
+    {% else %}
+      <a class="btn btn-danger" href="{% url 'event cancel signup'  event.id %}">Cancel</a>
```

Comment:
Maybe have an additional boolean check to only allow signups for future and upcoming events

Reply:
Agree with Bradley that you should only be able to sign up for upcoming events

Reply:
This also needs to be handled at the server level. Users can still just go to url.com/signup/1 if the link isn't there. This vulnerability is also present in edit events. Any user can edit any event with the right URL

Reply:
Made change to the template, not to the server. Users should still be able to retroactively cancel events they flaked on or mark attendance at event after the date if they didn't use our site to find the event. Maybe we should add a 'request' flow where any changes to the attendance after the date must be approved by the author.

Reply:
The method I used was pretty cool. In the model definition, add a function with the @property decorator. Then you can use model.propName in the HTML template files. It calls a calculated field instead of a piece of data in our database.

Suppose we have a datatype that stores kilometers. Then we can add
@property
def miles(self):
  return 1.6 * self.kilometers

Then we only need to store 1 distance and every time we want miles, its just calculated off kilometers.

Reply:
This does introduce some kind of vulnerability. Say 2 people are hosting competing events. One can scatter malicious links to cancel signups for the other event or signup for their event. Really we should have verification that anyone coming to that link is coming from our own site (and a valid button on our own site, they can still create links in other events). Not sure the best way to do that.